x-pack/plugins/security_solution/common/detection_engine/schemas/common/rule_monitoring.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import * as t from 'io-ts';
+import { enumeration, IsoDateString, PositiveInteger } from '@kbn/securitysolution-io-ts-types';
+
+// -------------------------------------------------------------------------------------------------
+// Rule execution status
+
+/**
+ * Custom execution status of Security rules that is different from the status
+ * used in the Alerting Framework. We merge our custom status with the
+ * Framework's status to determine the resulting status of a rule.
+ */
+export enum RuleExecutionStatus {
+  'succeeded' = 'succeeded',
+  'failed' = 'failed',
+  'going to run' = 'going to run',
+  'partial failure' = 'partial failure',
+  /**
+   * @deprecated 'partial failure' status should be used instead
+   */
+  'warning' = 'warning',
+}
+
+export const ruleExecutionStatus = enumeration('RuleExecutionStatus', RuleExecutionStatus);
+
+export const ruleExecutionStatusOrder = PositiveInteger;
+export type RuleExecutionStatusOrder = t.TypeOf<typeof ruleExecutionStatusOrder>;
+
+export const ruleExecutionStatusOrderByStatus: Record<
+  RuleExecutionStatus,
+  RuleExecutionStatusOrder
+> = {
+  [RuleExecutionStatus.succeeded]: 0,
+  [RuleExecutionStatus['going to run']]: 10,
+  [RuleExecutionStatus.warning]: 20,
+  [RuleExecutionStatus['partial failure']]: 20,
+  [RuleExecutionStatus.failed]: 30,
+};
+
+// -------------------------------------------------------------------------------------------------
+// Rule execution metrics
+
+export const durationMetric = PositiveInteger;
+export type DurationMetric = t.TypeOf<typeof durationMetric>;
+
+export const ruleExecutionMetrics = t.partial({
+  total_search_duration_ms: durationMetric,
+  total_indexing_duration_ms: durationMetric,
+  execution_gap_duration_s: durationMetric,
+});
+
+export type RuleExecutionMetrics = t.TypeOf<typeof ruleExecutionMetrics>;
+
+// -------------------------------------------------------------------------------------------------
+// Rule execution summary
+
+export const ruleExecutionSummary = t.type({
+  last_execution: t.type({
+    date: IsoDateString,
+    status: ruleExecutionStatus,
+    status_order: ruleExecutionStatusOrder,
+    message: t.string,
+    metrics: ruleExecutionMetrics,
+  }),
+});
+
+export type RuleExecutionSummary = t.TypeOf<typeof ruleExecutionSummary>;
+
+// -------------------------------------------------------------------------------------------------
+// Rule execution events
+
+export const ruleExecutionEvent = t.type({
+  date: IsoDateString,
+  status: ruleExecutionStatus,
```

Comment:
It corresponds to the current UI to reduce the scope of changes.

> How are we going to support other event types in the future? For example, we might want to log when a rule reaches max signals or other execution events.

Defining a proper read model for execution events and the final response schema was assumed to be done in https://github.com/elastic/kibana/issues/119598. What should be the response schema is a good question though 👍 I think it will depend on the UI requirements, and since they are not fully known, we can rely on the fact that the endpoint returning execution events is internal (we can introduce breaking changes between releases).

@spong are you going to work both on this ticket and https://github.com/elastic/kibana/issues/119599 at the same time or you'd need any help with the endpoint?

> Probably we need to rename ruleExecutionEvent to something more specific, like ruleStatusChangeEvent.

I can do that

Reply:
> @spong are you going to work both on this ticket and #119599 at the same time or you'd need any help with the endpoint?

Yeah, I'll tackle both as part of #119599 👍